solidity/test/router.test.ts

```diff
@@ -46,191 +39,146 @@ describe.skip('Router', async () => {
   beforeEach(async () => {
     const mailboxFactory = new TestMailbox__factory(signer);
     mailbox = await mailboxFactory.deploy(origin);
-    igp = await new TestInterchainGasPaymaster__factory(signer).deploy(
-      nonOwner.address,
-    );
+    igp = await new TestInterchainGasPaymaster__factory(signer).deploy();
     const requiredHook = await new TestMerkleTreeHook__factory(signer).deploy(
       mailbox.address,
     );
-    const defaultIsm = await new TestIsm__factory(signer).deploy();
+    defaultIsm = await new TestIsm__factory(signer).deploy();
     await mailbox.initialize(
       signer.address,
       defaultIsm.address,
       igp.address,
       requiredHook.address,
     );
-    router = await new TestRouter__factory(signer).deploy();
+    router = await new TestRouter__factory(signer).deploy(mailbox.address);
+    await router.initialize(igp.address, defaultIsm.address);
   });
 
   describe('#initialize', () => {
-    it('should set the mailbox', async () => {
-      await router.initialize(mailbox.address);
-      expect(await router.mailbox()).to.equal(mailbox.address);
+    it('should set the hook', async () => {
```

Comment:
Dunno if that's easy to do, but would like to test that overriding metadata and value works? Kind of implicit in GasRouter I guess but still

Reply:
it is in forge
will create an issue to migrate our router tests to forge and we will be rid of hardhat